contracts/governance/DeveloperNameGovernor.sol

```diff
@@ -0,0 +1,86 @@
+//SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.17;
+
+import { ECDSA } from "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+import { SignatureChecker } from "@openzeppelin/contracts/utils/cryptography/SignatureChecker.sol";
+
+import { IDeveloperRegistry } from "../interfaces/IDeveloperRegistry.sol";
+
+/**
+ * @title DeveloperNameGovernor
+ * @notice Contract that coordinates adding a new project for a Developer. Each Developer has their own DeveloperRegistrar which is associated
+ * with a .ers subnode in the ERS registry ([developer].ers). In order for a valid name claim the caller must submit a transaction with a valid
+ * signature signed by the coordinator of this contract over hash(developerAddress, developerName).
+ */
+
+contract DeveloperNameGovernor is Ownable {
+
+    using ECDSA for bytes;
+    using SignatureChecker for address;
+
+    /* ============ Events ============ */
+    event NameCoordinatorUpdated(address newNameCoordinator);
+
+    /* ============ State Variables ============ */
+    IDeveloperRegistry public immutable developerRegistry;
+    address public nameCoordinator;
+
+    /* ============ Constructor ============ */
+    /**
+     * @notice Constructor for DeveloperNameGovernor. Sets the owner and DeveloperRegistry.
+     *
+     * @param _developerRegistry    DeveloperRegistry contract
+     * @param _nameCoordinator      Address of the signer of name claims
+     */
+    constructor(IDeveloperRegistry _developerRegistry, address _nameCoordinator)
+        Ownable()
+    {
+        developerRegistry = _developerRegistry;
+        nameCoordinator = _nameCoordinator;
+    }
+
+    /* ============ External Functions ============ */
+
+    /**
+     * @notice Claim a name for a developer. The caller must submit a transaction with a valid signature signed by the name coordinator. The
+     * name coordinator performs checks for name validity and availability.
+     *
+     * @param _developerOwner       Address of the developer owner
+     * @param _developerName        ERS name of the developer
+     * @param _nameApprovalProof    Signature of the name coordinator over hash(developerOwner, developerName)
+     */
+    function claimName(
```

Comment:
I'm debating if we also want to validate the developer via a signature from their wallet -- or confirm that msg.sender is _developerOwner to assure intent. The downside is of course that you cannot delegate another cold wallet as the developer, but the upside is that it shows the developer is functional (e.g. can submit the tx so presumably can sign messages).

So to clarify two options:

- check msg.sender as developerOwner
- add in a signature that is validated from developerOwner so another account could theoretically broadcast

Slight preference for the former since it's a bit simply from a UX perspective, but obviously non-ideal if some really want to delegate an offline account as the developer (seems fairly unlikely at the moment)

Reply:
Reviewing a bit deeper here I see createNewDeveloperRegistrar needs the Developer to send a tx anyway, so let's just add a check here that msg.sender is the developerOwner.